### PR TITLE
Rover: added SYSID_ENFORCE parameter

### PR DESCRIPTION
**Pull request by khancyr**
As in copter and plane, add sysid_enfore to rover

### CODE REVIEW

File: APMrover2/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -1567,3 +1567,17 @@ void Rover::gcs_retry_deferred(void)
     gcs_send_message(MSG_RETRY_DEFERRED);
     GCS_MAVLINK::service_statustext();
 }
+
+/*
+  return true if we will accept this packet. Used to implement SYSID_ENFORCE
+ */
+bool GCS_MAVLINK_Rover::accept_packet(const mavlink_status_t &status, mavlink_message_t &msg)
+{
+  if (!rover.g2.sysid_enforce) {
```
Comment (gmorph) [inline APMrover2/GCS_Mavlink.cpp]: Indentation needs to be fixed in this file

File: APMrover2/Parameters.cpp [hunk 2]
```diff
@@ -570,6 +570,13 @@ const AP_Param::GroupInfo ParametersG2::var_info[] = {
     // @Path: ../libraries/AP_Stats/AP_Stats.cpp
     AP_SUBGROUPINFO(stats, "STAT", 1, ParametersG2, AP_Stats),
 
+    // @Group: SYSID_ENFORCE
+    // @DisplayName: GCS sysid enforcement
+    // @Description: This controls whether packets from other than the expected GCS system ID will be accepted
+    // @Values: 0:NotEnforced,1:Enforced
+    // @User: Advanced
+    AP_GROUPINFO("SYSID_ENFORCE", 11, ParametersG2, sysid_enforce, 0),
```
Comment (gmorph) [inline APMrover2/Parameters.cpp]: Why is this index 11?  Shouldn't it be 2?

**Comment by gmorph** [comment]
Great to have this in the code.  Just a couple of small comments I've made.  Once resolved we can merge it in.

File: APMrover2/Parameters.cpp (hunk 2 above)
Comment (khancyr) [inline APMrover2/Parameters.cpp]: I take it from copter. But I can change it to 2

**Comment by khancyr** [comment]
@gmorph Done ! I take the opportunity to fixe indentation on all the gcs_mavlink.cpp at once

**Pull request merged**